include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -210,46 +182,23 @@ class sequence_file_input_format_REMOVEME<format_embl>
         ++stream_it;
         ++stream_it;
     }
-};
-
-//!\brief The seqan3::sequence_file_output_format_REMOVEME specialisation that can write formatted EMBL.
-//!\ingroup sequence
-template <>
-class sequence_file_output_format_REMOVEME<format_embl>
-{
-public:
-    //!\brief Exposes the format tag that this class is specialised with.
-    using format_tag = format_embl;
-
-    /*!\name Constructors, destructor and assignment
-     * \{
-     */
-    sequence_file_output_format_REMOVEME()                                                noexcept = default; //!< Defaulted.
-    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
-    sequence_file_output_format_REMOVEME(sequence_file_output_format_REMOVEME const &)                      = delete;
-    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
-    sequence_file_output_format_REMOVEME & operator=(sequence_file_output_format_REMOVEME const &)          = delete;
-    sequence_file_output_format_REMOVEME(sequence_file_output_format_REMOVEME &&)                  noexcept = default; //!< Defaulted.
-    sequence_file_output_format_REMOVEME & operator=(sequence_file_output_format_REMOVEME &&)      noexcept = default; //!< Defaulted.
-    ~sequence_file_output_format_REMOVEME()                                               noexcept = default; //!< Defaulted.
-    //!\}
 
-    //!\copydoc sequence_file_output_format::write
+    //!\copydoc sequence_file_output_format::write_sequence_record
     template <typename stream_type,     // constraints checked by file
               typename seq_type,        // other constraints checked inside function
               typename id_type,
               typename qual_type>
-    void write(stream_type                          & stream,
-               sequence_file_output_options const   & options,
-               seq_type                             && sequence,
-               id_type                              && id,
-               qual_type                            && SEQAN3_DOXYGEN_ONLY(qualities))
+    void write_sequence_record(stream_type                          & stream,
+                               sequence_file_output_options const   & options,
+                               seq_type                             && sequence,
+                               id_type                              && id,
+                               qual_type                            && SEQAN3_DOXYGEN_ONLY(qualities))
     {
         seqan3::ostreambuf_iterator stream_it{stream};
         [[maybe_unused]] size_t sequence_size = 0;
         [[maybe_unused]] char buffer[50];
         if constexpr (!detail::decays_to_ignore_v<seq_type>)
-            sequence_size = ranges::size(sequence);
+            sequence_size = std::ranges::distance(sequence);
```

Comment:
Are non sized sequences allowed?

Reply:
yes I never require sized ranges. It might be inconsistent in the code because I did not have this in the back of my mind when writing this code mnonths ago.

Reply:
Hm, it seems like we should definitely document clearly what the requirements are...